filebeat/input/udp/input.go

```diff
@@ -36,33 +42,54 @@ func NewInput(
 		return nil, err
 	}
 
+	config := defaultConfig
+	if err = cfg.Unpack(&config); err != nil {
+		return nil, err
+	}
+
 	forwarder := harvester.NewForwarder(out)
+	callback := func(data []byte, metadata udp.Metadata) {
+		e := util.NewData()
+		e.Event = beat.Event{
+			Timestamp: time.Now(),
+			Meta: common.MapStr{
+				"truncated": metadata.Truncated,
+			},
+			Fields: common.MapStr{
+				"message": string(data),
+				"source":  metadata.RemoteAddr.String(),
+			},
+		}
+		forwarder.Send(e)
+	}
+
+	udp := udp.New(&config.Config, callback)
+
 	return &Input{
-		outlet:    out,
-		harvester: NewHarvester(forwarder, cfg),
-		started:   false,
+		outlet:  out,
+		udp:     udp,
+		started: atomic.MakeBool(false),
 	}, nil
 }
 
-// Run starts and execute the UDP server.
+// Run starts and start the UDP server and read events from the socket
 func (p *Input) Run() {
-	if !p.started {
-		logp.Info("Starting udp input")
-		p.started = true
-		go func() {
-			defer p.outlet.Close()
-			err := p.harvester.Run()
-			if err != nil {
-				logp.Err("Error running harvester:: %v", err)
-			}
-		}()
+	if !p.started.Load() {
```

Comment:
I _think_ we talked about this and you were going to open an issue for it. If you did/do can you add a link back to here.

There's a race condition problem with using an `atomic.Bool` to protect the underlying state of the server. A mutex needs to be used instead to protect it.

Reply:
Correct, let me switch that to a mutex.

Reply:
I will do the same change in the TCP.

Reply:
I've moved it to a mutex instead and will do a followup issue, I need to get a bit more insight about was is managing the input before proposing something. I did the same changes in the TCP.

Reply:
@andrewkroh ^